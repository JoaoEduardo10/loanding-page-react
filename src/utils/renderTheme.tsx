import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { GlobalStyled } from '../styles/globalStyles';
import { theme } from '../styles/theme';

export const renderTheme = (children: JSX.Element) => {
	return (
		render(
			<ThemeProvider theme={theme}>
				{children}
				<GlobalStyled />
			</ThemeProvider>
		)
	);
};