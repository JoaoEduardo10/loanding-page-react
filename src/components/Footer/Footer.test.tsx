import { screen } from '@testing-library/react';
import { Footer } from '.';
import { renderTheme } from '../../utils/renderTheme';

describe('<Footer />', () => {
	it('should render a Footer', () => {
		const { container } = renderTheme(<Footer html='<h1>Children</h1>' />);

		expect(screen.getByRole('heading', { name: 'Children' })).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});