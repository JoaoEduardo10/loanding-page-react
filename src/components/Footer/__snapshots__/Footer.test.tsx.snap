// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render a Footer 1`] = `
.c0 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  font-size: 2.4rem;
}

.c1 {
  text-align: center;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 1.6rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1>
          Children
        </h1>
      </div>
    </div>
  </div>
</div>
`;
